src/DropboxAdapter.php

```diff
@@ -182,11 +182,39 @@ public function listContents($directory = '', $recursive = false): array
             return [];
         }
 
-        return array_map(function ($entry) {
+        $cleanedPathDisplay = [];
```

Comment:
This is quite a long function. And to me it's isn't entirely clear what is happening. Your first reflex might be to add some comments here and there to make things more clear. But what I would prefer is that this block of code would be split up in multiple smaller functions that can be covered by unit tests.